app/Helpers/render/site-award.php

```diff
@@ -8,39 +8,72 @@
 
 function SeparateAwards(array $userAwards): array
 {
-    // Get all dev event game IDs in a single optimized query.
-    $devEventGameIds = GameSet::query()
-        ->whereHas('games', fn ($q) => $q->whereIn('game_id', array_column($userAwards, 'AwardData')))
```

Comment:
Prefiltering the list reduces the number of records in the `whereIn` clause, which provides a sizable performance increase.